### ISSUE TITLE
Deterministic label creation

### ISSUE THREAD

**Issue by iqml** [IQVIA-ML/TreeParzen.jl#18]
**Describe the solution you'd like**

This would mean that the user doesn't need to separately specify the label, such as
```julia
HP.Normal(:foo, 5.0, 7.0) # :foo is the label
```

We would need to create the labels such that we can re-read the history when it's gone to MLJ and come back again.

**Comment by kainkad** [comment]
After reviewing this issue in relation to #10, it will be also closed due to the following:

- Users can't see what labels they're creating (hides keys from `vals`)
- Diverges from Hyperopt: Hyperopt requires explicit labels; this adds implicit behaviour
- Harder to debug: If something goes wrong, users see `model__opt0__lr` in `trial.vals`, not `learning_rate` or whatever that autogenerated label would be
- Makes it harder to reason about what parameters exist in a space
